src/nbl/video/CVKLogicalDevice.h

```diff
@@ -124,51 +327,472 @@ class CVKLogicalDevice final : public ILogicalDevice
     {
         return;
     }
-            
+
+    bool bindBufferMemory(uint32_t bindInfoCount, const SBindBufferMemoryInfo* pBindInfos) override
+    {
+        for (uint32_t i = 0u; i < bindInfoCount; ++i)
+        {
+            if ((pBindInfos[i].buffer->getAPIType() != EAT_VULKAN) /*|| (pBindInfos[i].memory->getAPIType() != EAT_VULKAN)*/)
+                continue;
+
+            VkBuffer vk_buffer = static_cast<const CVulkanBuffer*>(pBindInfos[i].buffer)->getInternalObject();
+            VkDeviceMemory vk_memory = static_cast<const CVulkanMemoryAllocation*>(pBindInfos[i].memory)->getInternalObject();
+            if (vkBindBufferMemory(m_vkdev, vk_buffer, vk_memory, static_cast<VkDeviceSize>(pBindInfos[i].offset)) != VK_SUCCESS)
+                return false;
+        }
+
+        return true;
+    }
+
+    core::smart_refctd_ptr<IGPUBuffer> createGPUBuffer(
+        const IDriverMemoryBacked::SDriverMemoryRequirements& initialMreqs,
+        const bool canModifySubData = false) override
+    {
+        // Todo(achal): I would probably need to create an IGPUBuffer::SCreationParams
+        VkBufferCreateInfo vk_createInfo = { VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO };
+        vk_createInfo.pNext = nullptr; // Each pNext member of any structure (including this one) in the pNext chain must be either NULL or a pointer to a valid instance of VkBufferDeviceAddressCreateInfoEXT, VkBufferOpaqueCaptureAddressCreateInfo, VkDedicatedAllocationBufferCreateInfoNV, VkExternalMemoryBufferCreateInfo, VkVideoProfileKHR, or VkVideoProfilesKHR
+        vk_createInfo.flags = static_cast<VkBufferCreateFlags>(0); // Nabla doesn't support any of these flags
+        vk_createInfo.size = initialMreqs.vulkanReqs.size;
+        vk_createInfo.usage = initialMreqs.vulkanBufferUsageFlags;
+        vk_createInfo.sharingMode = static_cast<VkSharingMode>(initialMreqs.sharingMode); 
+
+        if (initialMreqs.queueFamilyIndices)
+        {
+            vk_createInfo.queueFamilyIndexCount = initialMreqs.queueFamilyIndices->size();
+            vk_createInfo.pQueueFamilyIndices = initialMreqs.queueFamilyIndices->data();
+        }
```

Comment:
I dont like `queueFamilyIndices` being a smart refctd ptr, a pointer+count would have worked just fine (same API as vulkan)

Also you dont initialize these variables (you dont have an `else` case)

Reply:
resolved